woocommerce/woocommerce-ios
# Order Creation: Add shipping line UI

Closes: https://github.com/woocommerce/woocommerce-ios/issues/6024

## Description

This PR adds navigation to the fully functional "Shipping Line Details" screen from the new order screen.

## Changes

- Adds `ShippingLineDetails` view.
- Adds `shippingMethodTitle` to `PaymentDataViewModel`.
- Adds `shippingMethodTitle` in the Payment section
- Connects navigation and data flow from the `ShippingLineDetails` view.
- Extracts `PriceFieldFormatter` from `SimplePaymentsAmountViewModel`.
- Tests for `PriceFieldFormatter` are mostly duplicated from `SimplePaymentsAmountViewModel` tests.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "Add Product" button and select any product to enable "Payment" section.
5. Tap "Add shipping" button.
6. Check the Shipping Line Details screen, enter both amount and title.
7. Tap "Done" to save.
8. Confirm that the "Payment" section displays correct shipping line data.
9. Confirm that the "Order Total" is correctly updated to display products + shipping price.
10. Tap the "Shipping" row to open Shipping Line Details.
11. Confirm that fields are prefilled with correct data.
12. Tap "Delete Shipping from Order".
13. Confirm that the shipping row is replaced with a button.
14. Confirm that the "Order Total" is correctly updated to display products price.

## Video

https://user-images.githubusercontent.com/3132438/154549619-deff6c2d-d313-49b4-a47e-c59eb5447539.mp4

---
